eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -1,741 +1,512 @@
-    #                fit_params.get(i).max) for i in fit_params]
-
-    # Create new model with best fit parameters
-    # params = Parameters()
-
-    # Try to store each as an attribute
-    # for param in new_params:
-    #     setattr(params, param[0], param[1:])
-
+    
     # Make a new model instance
     best_model = copy.copy(model)
     best_model.components[0].update(fit_params, freenames)
-    # best_model.parameters = params
-    # best_model.name = ', '.join(['{}:{}'.format(k, round(v[0], 2)) for k, v in params.dict.items()])
-
+    
     model.update(fit_params, freenames)
     model_lc = model.eval()
     residuals = (lc.flux - model_lc) #/ lc.unc
-
-    if meta.isplots_S5 > 1:
-        model.plot(time=lc.time, draw=True)
-
-        fig, ax = plt.subplots(2,1)
-        ax[0].errorbar(lc.time, lc.flux, yerr=lc.unc, fmt='.')
-        ax[0].plot(lc.time, model_lc, zorder = 10)
-
-
-        ax[1].errorbar(lc.time, residuals, yerr=lc.unc, fmt='.')
-        plt.savefig(meta.outputdir + 'figs/lc_dynesty.png', dpi=300)
-        plt.show()
-
-    ln_like_val = (-0.5 * (np.sum((residuals / lc.unc) ** 2 + np.log(2.0 * np.pi * (lc.unc) ** 2))))
-    chi2 = np.sum((residuals / lc.unc) ** 2)
-    chi2red = chi2 / (len(lc.unc))
-    if meta.run_verbose:
-        print(len(lc.unc))
-        print(chi2)
-        print(chi2red)
-    # best_model.parameters = params
-    # best_model.name = ', '.join(['{}:{}'.format(k, round(v[0], 2)) for k, v in params.dict.items()])
-    # Plot RMS vs. bin size looking for time-correlated noise
-
+    
+    # Plot fit
+    if meta.isplots_S5 >= 1:
+        plot_fit(lc, model, meta, fitter='dynesty')
+    
+    # Compute reduced chi-squared
+    chi2red = computeRedChiSq(lc, model, meta, freenames)
+    
     print('\nDYNESTY RESULTS:\n')
     for freenames_i, fit_params_i in zip(freenames, fit_params):
         print('{0}: {1}'.format(freenames_i, fit_params_i))
-
-    # PLOT ALLAN PLOT
-    if meta.isplots_S5 > 3:
-        rmsplot(lc,model_lc, meta, figname='allanplot_dynesty.png')
+    
+    # Plot Allan plot
+    if meta.isplots_S5 >= 3:
+        plot_rms(lc, model, meta, fitter='dynesty')
+    
+    best_model.__setattr__('chi2red',chi2red)
+    best_model.__setattr__('fit_params',fit_params)
 
     return best_model
 
-
-
-
-def lmfitter(time, data, model, meta, uncertainty=None, **kwargs):
-    """Use lmfit
+def lmfitter(lc, model, meta, **kwargs):
+    """Perform a fit using lmfit.
 
     Parameters
     ----------
-    data: sequence
-        The observational data
-    model: ExoCTK.lightcurve_fitting.models.Model
-        The model to fit
-    uncertainty: np.ndarray (optional)
-        The uncertainty on the (same shape) data
-    method: str
-        The name of the method to use
-    name: str
-        A name for the best fit model
-    verbose: bool
-        Print some stuff
+    lc: eureka.S5_lightcurve_fitting.lightcurve.LightCurve
```

Comment:
I believe the lmfitter wasn't implemented yet, and after a bit of playing around I can't get the parameters in the right format. Perhaps someone who has used lmfit before can fix this later? For now, I just removed it as a S5 fitting option in the template ecf file.

Reply:
Fair enough, I'm just about to test it and may come to the same conclusion

Reply:
Yup, lmfitter is currently still very broken - let's do as you say and hide it from users for now and we can work on resolving this later

Reply:
Not sure if I've ever gotten lmfit to work properly. 🤷‍♂️